### PR TITLE
Upgrade version 1.3.3.post1

### PR DESCRIPTION
**Pull request by 88manpreet**

### CODE REVIEW

File: kafka/client.py [hunk 1]
```diff
@@ -298,13 +298,7 @@ def failed_payloads(payloads):
...
 
             request = encoder_fn(payloads=broker_payloads)
-            # decoder_fn=None signal that the server is expected to not
-            # send a response.  This probably only applies to
-            # ProduceRequest w/ acks = 0
-            expect_response = (decoder_fn is not None)
-            if expect_response:
-                selector.register(conn._sock, selectors.EVENT_READ, conn)
-            future = conn.send(request, expect_response=expect_response)
+            future = conn.send(request)
```
Comment (ecanzonieri) [inline kafka/client.py]: I think this is partially reverting the change I made to use selectors. This change should not be reverted. I'm not sure how this code even works.

File: kafka/client.py [hunk 2]
```diff
@@ -741,6 +735,17 @@ def send_offset_request(self, payloads=[], fail_on_error=True,
                 if not fail_on_error or not self._raise_on_response_error(resp)]
 
     @time_metric('offset_commit')
```
Comment (ecanzonieri) [inline kafka/client.py]: Something is wrong here. This seems to be a new request, but the time_metric name is still matching the one of the method below.

File: docs/changelog.rst [hunk 3]
```diff
@@ -1,6 +1,165 @@
...
+* Add ClusterMetadata sphinx documentation (dpkp)
+
+Legacy Client
+-------------
+* Add send_list_offset_request for searching offset by timestamp (charsyam 1001)
+* Use select to poll sockets for read to reduce CPU usage (jianbin-wei 958)
+* Use select.select without instance bounding (adamwen829 949)
+
+
+1.3.2 (Dec 28, 2016)
```
Comment (neoeahit) [inline docs/changelog.rst]: Unrelated but does it make sense to update this file with our post versions updates going forward, mostly for tracking? @ecanzonieri @ny2ko what do you think

File: kafka/client.py [hunk 4]
```diff
@@ -741,6 +735,17 @@ def send_offset_request(self, payloads=[], fail_on_error=True,
...
+    def send_list_offset_request(self, payloads=[], fail_on_error=True,
+                            callback=None):
```
Comment (neoeahit) [inline kafka/client.py]: nit: the spacing seems off?

File: kafka/client.py [hunk 5]
```diff
@@ -298,13 +298,7 @@ def failed_payloads(payloads):
                 continue
 
             request = encoder_fn(payloads=broker_payloads)
-            # decoder_fn=None signal that the server is expected to not
-            # send a response.  This probably only applies to
-            # ProduceRequest w/ acks = 0
-            expect_response = (decoder_fn is not None)
-            if expect_response:
-                selector.register(conn._sock, selectors.EVENT_READ, conn)
```
Comment (ny2ko) [inline kafka/client.py]: I don't think we want to get rid of this part. Rather change to:

if request.expect_response():
    selector.register(conn._sock, selectors.EVENT_READ, conn)

File: docs/changelog.rst (hunk 3 above)
Comment (ny2ko) [inline docs/changelog.rst]: Sounds like a plan to me. Additional merge conflict when merging but should be easily handled

File: kafka/conn.py [hunk 6]
```diff
@@ -86,76 +163,8 @@ class BrokerConnection(object):
...
-            sasl_mechanism (str): string picking sasl mechanism when security_protocol
-                is SASL_PLAINTEXT or SASL_SSL. Currently only PLAIN is supported.
-                Default: None
-            sasl_plain_username (str): username for sasl PLAIN authentication.
-                Default: None
-            sasl_plain_password (str): passowrd for sasl PLAIN authentication.
-                Defualt: None
-        """
-        self.host = host
         self.hostname = host
```
Comment (88manpreet) [inline kafka/conn.py]: Checked. Both are needed.

File: kafka/client.py [hunk 7]
```diff
@@ -735,9 +735,9 @@ def send_offset_request(self, payloads=[], fail_on_error=True,
         return [resp if not callback else callback(resp) for resp in resps
                 if not fail_on_error or not self._raise_on_response_error(resp)]
 
-    @time_metric('offset_commit')
+    @time_metric('list_offset')
```
Comment (neoeahit) [inline kafka/client.py]: can we call this offset_list to be consistent with naming of other metrics in this class?

File: pylint.rc [hunk 8]
```diff
@@ -1,2 +1,3 @@
 [TYPECHECK]
 ignored-classes=SyncManager,_socketobject
+disable=E1129
```
Comment (ny2ko) [inline pylint.rc]: Can you verify that this issue affects upstream too

Comment (88manpreet) [inline pylint.rc]: Verified that this issue affects upstream too.

**Pull request merged**